src/common/src/cache.rs

```diff
@@ -766,25 +768,33 @@ impl<K: LruKey + Clone, T: LruValue> LruCache<K, T> {
     ) -> Result<Result<CachableEntry<K, T>, E>, RecvError>
     where
         F: FnOnce() -> VC,
-        E: Error,
-        VC: Future<Output = Result<(T, usize), E>>,
+        E: Error + Send + 'static,
+        VC: Future<Output = Result<(T, usize), E>> + Send + 'static,
     {
         match self.lookup_for_request(hash, key.clone()) {
             LookupResult::Cached(entry) => Ok(Ok(entry)),
             LookupResult::WaitPendingRequest(recv) => {
                 let entry = recv.await?;
                 Ok(Ok(entry))
             }
-            LookupResult::Miss => match fetch_value().await {
-                Ok((value, charge)) => {
-                    let entry = self.insert(key, hash, charge, value);
-                    Ok(Ok(entry))
-                }
-                Err(e) => {
-                    self.clear_pending_request(&key, hash);
-                    Ok(Err(e))
-                }
-            },
+            LookupResult::Miss => {
+                let this = self.clone();
```

Comment:
could we rewrite it as 
```rust
let ret = tokio::spawn(fetch_value).await;
```

Reply:
Nope. fetch_value itself is not 'static. The future it returns is 'static.